packages/coordinator/src/components/Popup/Popup.tsx

```diff
@@ -0,0 +1,68 @@
+import { noop } from "lodash";
+import {
+  Dialog,
+  DialogTitle,
+  DialogActions,
+  DialogContent,
+  makeStyles,
+} from "@material-ui/core";
+import Button, { ButtonVariant } from "../Button";
+
+export type PopupProps = {
+  open: boolean;
+  title?: string | React.ReactNode;
+  content?: string | React.ReactNode;
+  onClose: (event: object, reason: string) => void;
+
+  // Buttons
+  primaryButtonText?: string;
+  onPrimaryButtonClick?: () => void;
+  cancelButtonText?: string;
+  onCancelButtonClick?: () => void;
+};
+
+export function Popup(props: PopupProps) {
+  const {
+    open,
+    title,
+    content,
+    onClose,
+    primaryButtonText,
+    onPrimaryButtonClick,
+    cancelButtonText,
+    onCancelButtonClick,
+  } = props;
+
+  const centerStyle = makeStyles({
+    root: {
+      justifyContent: "center",
+      textAlign: "center",
+    },
+  });
+
+  const classes = centerStyle();
+
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <DialogTitle>{title}</DialogTitle>
+      <DialogContent classes={classes}>{content}</DialogContent>
+      <DialogActions classes={classes}>
+        {primaryButtonText && (
+          <Button
+            title={primaryButtonText}
+            onClick={onPrimaryButtonClick || noop}
+            color={"secondary"}
+          />
+        )}
+        {cancelButtonText && (
+          <Button
+            title={cancelButtonText}
+            onClick={onCancelButtonClick || noop}
```

Comment:
Why is the `noop` needed?

Reply:
`onCancelButtonClick` is optional and without `noop`, the linter is mad.
So instead of creating the props for button in a separate code and check  `onCancelButtonClick`'s condition, we will pass `noop` , I think it is more readable.

Reply:
Got it, cool